plugins/common/tls/config.go

```diff
@@ -81,6 +82,15 @@ func (c *ClientConfig) TLSConfig() (*tls.Config, error) {
 		}
 	}
 
+	if c.TLSMinVersion != "" {
+		version, err := ParseTLSVersion(c.TLSMinVersion)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"could not parse tls min version %q: %v", c.TLSMinVersion, err)
+		}
+		tlsConfig.MinVersion = version
```

Comment:
Thanks for picking this up! I've not reviewed the whole PR but thought I'd mention that 1.0 and 1.1 are now widely deprecated (all the browsers, twitter, etc, etc) and so I suggest logging a warning if a) the min version is less than 1.2 or b) the default tls version from Go is less than 1.2[1]. Since telegraf 1.x will (presumably) continue to use the Go defaults, this has the added benefit of discoverability for people upgrading. It will continue to be useful in 2.x when it (presumably) sets the min version to 1.2 since users choosing a lower version will be reminded they are choosing an insecure configuration.

[1] https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/crypto/tls/common.go;l=685 - clients default to 1.2 but servers default to 1.0 so runtime detecting would future-proof this.

Reply:
In https://github.com/influxdata/telegraf/issues/8171#issuecomment-1183270132 I mentioned that 1.2 is used as a client and 1.0 is used as a server. This means that today, telegraf is enforcing 1.2 for all plugins that are acting as HTTP clients. I wonder if it makes sense to consider defaulting to 1.2 in telegraf 1.x? There is regression potential for listener plugins that would be acting as servers and defaulting to 1.0 as well as for people who might be working around the newer golang and choosing to compile telegraf with an older golang. The number of users relying on 1.0/1.1 in their telegraf listener or compiling with an older golang seems low... not to mention this PR provides the necessary escape hatch.

Fine to wait to set the default to 1.2 in 2.x, but I hadn't considered the recent Golang changes and that telegraf has already effectively changed its default for the vast majority of its users already.

Reply:
Hmmm maybe I'm missing something, but as we do not set the TLS minimum version up-to-now, we default to Golang's default which is TLS v1.2 since go 1.18.
What is your suggestion here?

Reply:
v1.2 is the minimum default for only clients. v1.0 is the minimum for servers (https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/crypto/tls/common.go;l=685). You're right that most of the code today is getting the 1.2 minimum version (since most of the plugins are operating as tls client), but there is some code operating as a server. My suggestion (or rather pondering) is since the upgrade to 1.18 appeared to go fine, perhaps pull the trigger in telegraf 1.x for setting 1.2 explicitly so we cover our tls server code too.

Reply:
Pushed for making v1.2 the (explicit) default for both server and client. Allowing to revert on a plugin basis...